runtimes/peregrine/src/lib.rs

```diff
@@ -937,6 +943,91 @@ impl pallet_proxy::Config for Runtime {
 	type WeightInfo = weights::pallet_proxy::WeightInfo<Runtime>;
 }
 
+parameter_types! {
+	pub CheckingAccount: AccountId = PolkadotXcm::check_account();
+}
+
+impl pallet_asset_swap::Config for Runtime {
+	const POOL_ADDRESS_GENERATION_ENTROPY: [u8; 8] = *b"plt/eplt";
+
+	type AccountIdConverter = AccountId32ToAccountId32JunctionConverter;
+	type AssetTransactor = FungiblesAdapter<
+		Fungibles,
+		MatchesSwapPairXcmFeeFungibleAsset<Runtime>,
+		LocationToAccountIdConverter,
+		AccountId,
+		NoChecking,
+		CheckingAccount,
+	>;
+	type FeeOrigin = EnsureRoot<AccountId>;
+	type LocalCurrency = Balances;
+	type PauseOrigin = EnsureRoot<AccountId>;
+	type RuntimeEvent = RuntimeEvent;
+	type SubmitterOrigin = EnsureSigned<AccountId>;
+	type SwapOrigin = EnsureRoot<AccountId>;
+	type XcmRouter = XcmRouter;
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+pub struct NoopBenchmarkHelper;
+
+#[cfg(feature = "runtime-benchmarks")]
+impl pallet_assets::BenchmarkHelper<MultiLocation> for NoopBenchmarkHelper {
+	fn create_asset_id_parameter(id: u32) -> MultiLocation {
+		MultiLocation {
+			parents: 0,
+			interior: xcm::prelude::Here,
+		}
+	}
+}
```

Comment:
Yes, this PR is actually still a draft. I will fix everything else once we have the final version of the pallet, since this PR will keep rebasing. But good point, I'll leave the comment open for future reference.

Reply:
Yes, it is quite difficult not to get lost in details. sorry

Reply:
Actually this PR is supposed to be merged, so I moved those into `runtime-common`: https://github.com/KILTprotocol/kilt-node/pull/659/commits/9c4c8140f188539820abd655eb63f76b583fe7b5.